brayns/parameters/ApplicationParameters.cpp

```diff
@@ -104,11 +104,14 @@ void ApplicationParameters::parse(const po::variables_map& vm)
 {
     if (vm.count(PARAM_ENGINE))
     {
-        _engine = EngineType::ospray;
         const std::string& engine = vm[PARAM_ENGINE].as<std::string>();
-        for (size_t i = 0; i < sizeof(ENGINES) / sizeof(ENGINES[0]); ++i)
-            if (engine == ENGINES[i])
-                _engine = static_cast<EngineType>(i);
+        const size_t pos =
+            std::distance(ENGINES.begin(),
+                          std::find(ENGINES.begin(), ENGINES.end(), engine));
```

Comment:
Be careful with distance, it's complexity is linear even for std::vector. If you know you have a vector is much better to do iterator subtraction.

In this case, I woundn't even use it. Replace pos != ENGINES.size() with pos != ENGINES.end() and pos with *pos.

Reply:
I am using a map now to avoid all indexing and type casting.